infra/gcp/namespaces/ensure-namespaces.sh

```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/env bash
+#
+# Copyright 2019 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script creates namespaces, applies roles and role bindings and
+# checks if assigned role binding is the only one in the namespace
+# for k8s-infra projects
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+SCRIPT_DIR=$(dirname "${BASH_SOURCE[0]}")
+. "${SCRIPT_DIR}/lib.sh"
+
+role_filename="namespace-user-role.yml"
+role_binding_filename="namespace-user-role-binding.yml"
+role_binding_name_postfix="-namespace-user-binding"
+
+function usage() {
+    echo "usage: $0" > /dev/stderr
+    echo > /dev/stderr
+}
+
+if [ $# != 0 ]; then
+    usage
+    exit 1
+fi
+
+function create_namespace() {
+  kubectl create "$1"
+}
+
+function apply_role() {
+  role=$(sed -e "s/{{namespace}}/$1/" "$role_filename")
+
+  kubectl apply -f "$role"
```

Comment:
I don't think this works?  -f wants a filename and you are giving it the content.  You can either do:

`sed ... | kubectl apply -f  -` 

... or save to a temp file.  If you use stdin as above, you'll want to echo something useful - kubectl won't tell you much

Reply:
for pedanticness, please use the `-n "${namespace}"` flag - that way if it doesnt match the YAML, kubectl can yell.

Reply:
ack

Reply:
Fixed